api/server.go

```diff
@@ -42,55 +43,80 @@ func writeResponse(w http.ResponseWriter, status int, response *Response) {
 }
 
 func (a *API) createLoadAgentHandler(w http.ResponseWriter, r *http.Request) {
-	var config loadtest.Config
-	err := json.NewDecoder(r.Body).Decode(&config)
+	var data struct {
+		LoadTestConfig         loadtest.Config
+		SimpleControllerConfig *simplecontroller.Config `json:",omitempty"`
+		SimulControllerConfig  *simulcontroller.Config  `json:",omitempty"`
+	}
+	err := json.NewDecoder(r.Body).Decode(&data)
 	if err != nil {
 		writeResponse(w, http.StatusBadRequest, &Response{
-			Error: err.Error(),
+			Error: fmt.Sprintf("could not read request: %s", err),
 		})
 		return
 	}
-
-	if err := config.IsValid(); err != nil {
+	ltConfig := data.LoadTestConfig
+	if err := ltConfig.IsValid(); err != nil {
 		writeResponse(w, http.StatusBadRequest, &Response{
-			Error: err.Error(),
+			Error: fmt.Sprintf("could not validate config: %s", err),
 		})
 		return
 	}
-	logger.Init(&config.LogSettings)
 
-	agentId := r.FormValue("id")
-	if a.agents[agentId] != nil {
-		writeResponse(w, http.StatusBadRequest, &Response{
-			Error: fmt.Sprintf("load-test agent with id %s already exists", agentId),
-		})
-		return
-	}
+	logger.Init(&ltConfig.LogSettings)
 
 	var ucConfig control.Config
-	switch config.UserControllerConfiguration.Type {
+	switch ltConfig.UserControllerConfiguration.Type {
 	case loadtest.UserControllerSimple:
-		// TODO: pass simplecontroller path appropriately
-		ucConfig, err = simplecontroller.ReadConfig("")
+		var scc *simplecontroller.Config
+		scc = data.SimpleControllerConfig
+		if scc == nil {
```

Comment:
I think we can simplify and directly check for `data.SimpleControllerConfig == nil`.
Same for the SimulController case below.